### ISSUE TITLE
OrderedQ implementation different from Wolfram Language

### ISSUE THREAD

**Issue by hrueter** [Mathics3/mathics-core#201]
### Observed behavior
```
In[1]:= OrderedQ[{2,1}]
Out[1]= OrderedQ[{2,1}]
```
### Expected behavior
```
In[1]:= OrderedQ[{2,1}]
Out[1]= False
```
### Background
In WL `OrderedQ` takes a list of elements as input. Instead Mathics implementation of `OrderedQ` expects two arguments, whose order is checked against each other.
### Setup
Linux Mint 20.3
Mathics 4.0.0 on CPython 3.8.10 (default, Nov 26 2021, 20:14:08)  using SymPy 1.8, mpmath 1.2.1, numpy 1.17.4
### Possible Fix
I rewrote the `OrderedQ` class to obtain the behavior expected from above. However this still compares only the first two elements in the list, where Mathematica would check all of them.
https://github.com/Mathics3/mathics-core/blob/9eb8b1a47e69eea40c4ecf9532525649c0163e53/mathics/builtin/structure.py#L291

```
class OrderedQ(Builtin):
    """
    <dl>
    <dt>'OrderedQ[{$a$, $b$}]'
        <dd>is 'True' if $a$ sorts before $b$ according to canonical
        ordering.
    </dl>

    >> OrderedQ[{a, b}]
     = True
    >> OrderedQ[{b, a}]
     = False
    """

    def apply(self, expr, evaluation):
        "OrderedQ[expr_]"

        if expr.leaves[0] <= expr.leaves[1]:
            return SymbolTrue
        else:
            return SymbolFalse
```

**Comment by adamantinum** [comment]
@hrueter adding a loop to your fix should have implemented the function correctly

**Comment by TiagoCavalcante** [comment]
@hrueter this was fixed in #207 (thanks @adamantinum!)
